packages/cli/src/lib/implementation/only-plugins.middleware.unit.test.ts

```diff
@@ -0,0 +1,97 @@
+import { describe, expect, vi } from 'vitest';
+import { onlyPluginsMiddleware } from './only-plugins.middleware';
+
+vi.mock('@code-pushup/core', async () => {
+  const { CORE_CONFIG_MOCK }: typeof import('@code-pushup/test-utils') =
+    await vi.importActual('@code-pushup/test-utils');
+  const core: object = await vi.importActual('@code-pushup/core');
+  return {
+    ...core,
+    readRcByPath: vi.fn().mockResolvedValue(CORE_CONFIG_MOCK),
+    autoloadRc: vi.fn().mockResolvedValue(CORE_CONFIG_MOCK),
+  };
+});
+
+describe('onlyPluginsMiddleware', () => {
+  it('should forward equal values if not set', () => {
+    expect(
+      onlyPluginsMiddleware({
+        plugins: [],
+        categories: [],
+      }),
+    ).toStrictEqual({
+      plugins: [],
+      categories: [],
+    });
+  });
+
+  it('should filter plugins', () => {
```

Comment:
The test case descriptions are very vague. Is this a case where all plugins are filtered out, some, or none? These should be in dedicated test cases.

Reply:
added proper description